src/components/container/BloodPressureVisualization/BloodPressureVisualization.tsx

```diff
@@ -0,0 +1,303 @@
+            bpMetrics.maxSystolicAlert = Category[cat];
+            bpMetrics.maxSystolicAlertClass = cat === Category.Normal ? "mdhui-blood-pressure-metric-normal" : "mdhui-blood-pressure-metric-not-normal";
+        }
+        
+        return bpMetrics;
+    }
+
+    function buildDetailBlock( description : string, alert? : string, alertClass? : string, value? : number ){
+        return <div className="mdhui-blood-pressure-metrics-block">
+            <div className="mdhui-blood-pressure-metric-value">{value ?? "--"} <span className="mdhui-blood-pressure-units">mm HG</span></div>
+            <div className="mdhui-blood-pressure-metric-description">{description}</div>
+            <div className={alertClass}>{alert ?? "No data yet"}</div>
+        </div>
+    }
+
+    function buildInterval(entries : number[]){
+        const min = Math.min(...entries);
+        const closedInterval : ClosedInterval = {values : [min]};
+        
+        var max = min;
+        if (entries.length > 1){
+            max = Math.max(...entries);
+        }
+        closedInterval.values.push(max);
+        return closedInterval;
+    }
+
+    function assignClass(avgSystolic : number, avgDiastolic : number){
+
+        const systolicCategory = getSystolicCategory(avgSystolic);
+        const diastolicCategory = getDiastolicCategory(avgDiastolic);
+
+        if (systolicCategory === Category.Normal && diastolicCategory === Category.Normal){
+            return DumbbellClass["mdhui-dumbbell-in-range"];
+        }
+
+        return DumbbellClass["mdhui-dumbbell-out-of-range"];
+    }
+
+    function getSystolicCategory(systolicValue : number) {
+        if (systolicValue < 90){
+            return Category.Low
+        }
+
+        if (systolicValue >= 90 && systolicValue <= 119){
+            return Category.Normal;
+        }
+
+        if (systolicValue >= 120 && systolicValue <= 129){
+            return Category.Elevated;
+        }
+
+        if (systolicValue >= 130 && systolicValue <= 139){
+            return Category["Stage 1"];
+        }
+
+        if (systolicValue >= 140 && systolicValue <= 180){
+            return Category["Stage 2"];
+        }
+
+        if (systolicValue > 180){
+            return Category.Crisis;
+        }
+
+        return Category.Unknown;
+    }
+
+    function getDiastolicCategory(diastolicValue : number) {
+        if (diastolicValue < 60){
+            return Category.Low
+        }
+
+        if (diastolicValue >= 60 && diastolicValue <= 79){
+            return Category.Normal;
+        }
+
+        if (diastolicValue >= 80 && diastolicValue <= 89){
+            return Category["Stage 1"];
+        }
+
+        if (diastolicValue >= 90 && diastolicValue <= 120){
+            return Category["Stage 2"];
+        }
+
+        if (diastolicValue > 120){
+            return Category.Crisis;
+        }
+
+        return Category.Unknown;
+    }
+    
+    function initializeWeekPager() : Date {
```

Comment:
I think you can drop this and just call "getWeekStart" from here: https://github.com/CareEvolution/MyDataHelpsUI/blob/main/src/helpers/get-interval-start.ts#L15 with "Monday"

Though I would consider making the "weekStartsOn" a prop like this: https://github.com/CareEvolution/MyDataHelpsUI/blob/main/src/components/presentational/DateRangeCoordinator/DateRangeCoordinator.tsx#L9